packages/dd-trace/src/opentracing/propagation/text_map.js

```diff
@@ -168,6 +173,22 @@ class TextMapPropagator {
     return this._extractDatadogContext(parsed)
   }
 
+  _extractTraceparentContext (carrier) {
+    const headerValue = carrier[traceparentKey]
+    if (!headerValue) {
+      return null
+    }
+    const matches = headerValue.match(traceparentExpr)
+    if (matches.length) {
+      return new DatadogSpanContext({
```

Comment:
Since the trace context standard also includes a `tracestate` header that could transport other information specific to Datadog, could it be included as well?

Reply:
It could be indeed but what would it be? As stated in the ref:

> The main purpose of the tracestate HTTP header is to provide additional vendor-specific trace identification information across different distributed tracing systems and is a companion header for the traceparent field.

Could you lead me on what DD would store in this? Because it could be anything like:
```
tracestate: DD=XXXXXXXXX
```

Reply:
It could be used for many things, but in the short term it would mostly be the sampling priority and the origin. These are important for us to make consistent sampling decisions across the distributed trace to avoid incomplete traces. The traceparent sampling flag is not enough for our purpose because we have sampling priorities beyond "keep" or "drop".

I'll have to check internally if we have an existing spec for this because it needs to be the done the same way in every language. If we have one I'll let you know what the format is. If we don't then I think it's fine to leave them out for now, but then the feature would need to be enabled behind an experimental flag in the tracer configuration since it will not be complete. I'll update this thread as soon as I have more information, but if you want to go the safe route, I'd say to add the experimental config option right away and the tracestate header can then be added later in another PR. I think however that injection should still be added right away so that the tracer is fully compatible with W3C.